manager/usermanager/createUser.go

```diff
@@ -88,6 +90,8 @@ func CreateSocialUser(userStore *store.UserStore, user *models.UserCredentials)
 	for err != errors.ErrInvalidUser {
 		user.UserName = generateUserName(user.Name)
 		_, err = GetUserByUserName(userStore, user.UserName)
+		// A 100ms sleep will help the CPU to context-switch for other requests in case this loop becomes an infinite loop.
+		time.Sleep(time.Millisecond * 100)
```

Comment:
Ideally, sleep should have been between the 1st and 2nd invocation, 2nd and 3rd, 3rd and 4th, and so on. As per this code, there is no sleep between 1st and 2nd, but it is present between 2nd and 3rd, 3rd and 4th, and so on.

Not a big deal, but a point to note for future.